components/centraldashboard/app/api.ts

```diff
@@ -107,71 +111,97 @@ export class Api {
+                const cm = await this.k8sService.getExistingSharesConfigMap(req.params.namespace);
+                res.json(cm.data);
+            }catch(e){
+                return apiError({
+                    res, code: 500,
+                    error: ERRORS.invalid_get_existing_shares,
+                });
+            }
+            })
+        .get(
+            '/get-requesting-shares/:namespace',
+            async (req: Request, res: Response) => {
                 try {
-                  const cm = await this.k8sService.getUserFilerConfigMap(req.params.namespace);
-                  res.json(cm.data);
+                    const cm = await this.k8sService.getRequestingSharesConfigMap(req.params.namespace);
+                    res.json(cm.data);
                 }catch(e){
-                  return apiError({
-                    res, code: 500,
-                    error: ERRORS.invalid_get_user_filers,
-                  });
+                    return apiError({
+                        res, code: 500,
+                        error: ERRORS.invalid_get_requesting_shares,
+                    });
                 }
-              })
-            .patch(
-              '/update-filer/:namespace',
-              async (req: Request, res: Response) => {
+            })
+        .patch(
+            '/update-requesting-shares/:namespace',
+            async (req: Request, res: Response) => {
                 try {
-                  const cm = await this.k8sService.updateUserFilerConfigMap(req.params.namespace, req.body);
-                  res.json(cm.data);
+                    const cm = await this.k8sService.updateRequestingSharesConfigMap(req.params.namespace, req.body, req.user.email);
+                    res.json(cm.data);
                 }catch(e){
-                  return apiError({
-                    res, code: 500,
-                    error: ERRORS.invalid_update_filer,
-                  });
+                    return apiError({
+                        res, code: 500,
+                        error: ERRORS.invalid_update_requesting_shares,
+                    });
+                }
+            })
+        .patch(
+            '/update-existing-shares/:namespace',
+            async (req: Request, res: Response) => {
+                try {
+                    const cm = await this.k8sService.updateExistingSharesConfigMap(req.params.namespace, req.body);
+                    res.json(cm.data);
+                }catch(e){
+                    return apiError({
+                        res, code: 500,
+                        error: ERRORS.invalid_update_existing_shares,
+                    });
                 }
-              })
-              .delete(
-                '/delete-filer/:namespace',
-                async (req: Request, res: Response) => {
-                  try {
-                    await this.k8sService.deleteUserFilerConfigMap(req.params.namespace);
+            })
+        .delete(
```

Comment:
how would this api call be used? :?

Reply:
https://github.com/StatCan/kubeflow/pull/215/files/7fe3b9abbb0980b241f8e8082fb091b630731190#diff-7163f508162f4fae8b88e5d727e68869921bb99817073f3c82c0379a188c12d8R201

It gets called by the manage filers page, when a user deletes their only filer share, we simply delete the existing-shares CM instead of updating it to have no values.